Source/WebKit/GPUProcess/webrtc/LibWebRTCCodecsProxy.mm

```diff
@@ -133,12 +133,13 @@
         if (frameRateMonitor)
             frameRateMonitor->update();
 
-        auto videoFrame = WebCore::VideoFrameCV::create(MediaTime(timeStampNs, 1), false, WebCore::VideoFrame::Rotation::None, pixelBuffer);
+        auto newFrame = WebCore::VideoFrameCV::create(MediaTime(timeStampNs, 1), false, WebCore::VideoFrame::Rotation::None, pixelBuffer);
         if (resourceOwner)
-            videoFrame->setOwnershipIdentity(resourceOwner);
+            newFrame->setOwnershipIdentity(resourceOwner);
         if (videoFrameObjectHeap) {
-            auto properties = videoFrameObjectHeap->add(WTFMove(videoFrame));
-            connection->send(Messages::LibWebRTCCodecs::CompletedDecoding { identifier, timeStamp, timeStampNs, WTFMove(properties) }, 0);
+            auto properties = RemoteVideoFrameProxy::properties(newFrame);
+            auto [newFrameIdentifier, completion] = videoFrameObjectHeap->addWithCompletion(WTFMove(newFrame));
+            connection->sendWithAsyncReply(Messages::LibWebRTCCodecs::CompletedDecoding { identifier, timeStamp, timeStampNs, { newFrameIdentifier, WTFMove(properties) } }, WTFMove(completion), 0);
```

Comment:
I guess these changes are not strictly needed anymore for your web process changes.

That said, this still ties cameras/decoders thread with rendering threads for video frames.
Why not simply adding the video frame synchronously here and have the completion handler returned by the object heap to remove the frame if called with false (aka WebProcess adoption failed)?
Or split these changes in its own PR.